tests/autoyast/prepare_profile.pm

```diff
@@ -26,25 +26,41 @@ use autoyast qw(
   expand_version
   adjust_network_conf
   expand_variables
-  upload_profile);
+  upload_profile
+  get_test_data_files);
 
 sub run {
-    my $path = get_required_var('AUTOYAST');
-    # Get file from data directory
-    my $profile = get_test_data($path);
-
-    $path    = detect_profile_directory(profile => $profile, path => $path);
-    $profile = get_test_data($path);
-    die "Empty profile" unless $profile;
-
-    # if profile is a template, expand and rename
-    $profile = expand_template($profile) if $path =~ s/^(.*\.xml)\.ep$/$1/;
-    die $profile                         if $profile->isa('Mojo::Exception');
-
-    $profile = expand_version($profile);
-    $profile = adjust_network_conf($profile);
-    $profile = expand_variables($profile);
-    upload_profile(profile => $profile, path => $path);
+    my $files;
+    my $ay_path = get_required_var('AUTOYAST');
+
+    if ($ay_path =~ /\/$/) {
```

Comment:
As far as I understand, here we rely only on the last "/" in the path. So, case `AUTOYAST=path/to/folder` will not work. Maybe it is better to check if the path is a directory?

Something, like

```perl
if ( -d $ay_path ) {
    # When using rules and classes, multiple xml files will be processed
    $files = get_test_data_files($ay_path);
}
```

Reply:
I has to be written like that, is in the YaST documentation with the last '/', so if someone try that we might find some better way to handle, but it should only read one file and fail, here is the doc: https://documentation.suse.com/sles/15-SP3/html/SLES-all/rulesandclass.html

Reply:
Oh, I see. Could you please highlight that this slash in the end is important? At least a comment, but better a check with meaningful error message. Because otherwise it may confuse even us in future.

Right now if `AUTOYAST=path/to/file` will be provided, the code execution will go to `else` block and will try to find a file with that name. It may be hard to debug in future.

Reply:
that is exactly how is supposed to be, to point to a file and if it is the wrong one some error is expected (but of course that requires some knowledge of ay and features rules and classes), but I got your point, the only way to give an error on on particular path is to fix expectation as we just get a string, and the path is evaluated later in different ways. I wanted to be a bit conservative as the code is not new at old, it has been with us for years, but at the end I could separate in two files and reuse the common part in the loop and give the user/tester meaningful error when the settings is not as expected.

Reply:
I thought it is simple and expected we can do something like this:

```perl
elsif  ( -d $ay_path && $ay_path !~ /\/$/) {
   die "Folder path is used without "/" in the end. Please add "/" in case you want to use rules in AutoYaST"
}
```

Will that work?

Reply:
no, as I mentioned before it cannot work, because the $ay_path is a relative path that is resolved later to its full path, so -d will always tell us that doesn't exists. The worker is checking with Perl using -d if the file exists but at that point is just an string unprocessed, is another way to say it.